packages/venia-concept/src/actions/cart/__tests__/asyncActions.spec.js

```diff
@@ -529,24 +532,9 @@ test('updateItemInCart thunk dispatches special failure if guestCartId is not pr
         2,
         actions.updateItem.receive(error)
     );
-    // and now, the createGuestCart thunk
-    expect(dispatch).toHaveBeenNthCalledWith(3, expect.any(Function));
 });
 
 test('updateItemInCart tries to recreate a guest cart on 404 failure', async () => {
-    getState
-        .mockImplementationOnce(() => ({
-            cart: { guestCartId: 'OLD_AND_BUSTED' },
-            user: { isSignedIn: false }
-        }))
-        .mockImplementationOnce(() => ({
-            cart: { guestCartId: 'CACHED_CART' },
-            user: { isSignedIn: false }
-        }))
-        .mockImplementationOnce(() => ({
-            cart: {},
-            user: { isSignedIn: false }
-        }));
```

Comment:
`getState` is now only called once and `cart` and `user` are destructured from there.

The tweak to the retry logic made it so that we don't need this third mock implementation.